### PR TITLE
Add a System.DateTime generator

### PR DESCRIPTION
**Pull request by moodmosaic**

### CODE REVIEW

File: Jack/Gen.fs [hunk 1]
```diff
@@ -434,6 +434,15 @@ module Gen =
     let float : Gen<float> =
         double |> map float
 
+    let dateTime : Gen<System.DateTime> =
+        gen { let! y = range 1753 9999
```
Comment (moodmosaic) [inline Jack/Gen.fs]: If you're wondering about `1753` and `9999`, it's the date range in SQL Server ([source](http://stackoverflow.com/a/548373/467754)).

Comment (jystic) [inline Jack/Gen.fs]: Would it ever make sense to have a larger range? The gregorian calendar is reasonable from about 1600 onwards for example.

**Comment by jystic** [comment]
It might be nice to change the year generator (which is currently `range 1753 9999`) to something which shrinks to a "nice" year like 2000.

What I mean by that is, construct a custom `Gen` so that it still has the range `1753` to `9999` but when it shrinks, it shrinks towards `2000`, e.g.:
```fs
let! y = Gen.create (Shrink.towards 2000) (Random.range 1753 9999)
```

File: Jack/Gen.fs (hunk 1 above)
Comment (jystic) [inline Jack/Gen.fs]: tbh, I don't mind what this is, it could even be `0000` as that's the valid range for .NET DateTime I think?

**Comment by moodmosaic** [comment]
Thanks for the great feedback! You're right; we should actually support the whole range of System.DateTime

```
[00:00:00.0000000 UTC, January 1, 0001 — 23:59:59.9999999 UTC, December 31, 9999]
```

in the Gregorian calendar, and shrink to a "nice" year like `2000`.

**Comment by moodmosaic** [comment]
So I've just updated the pull request ‒ I've also added some (missing) comments via eeb7391 :sweat_smile:

**Comment by jystic** [comment]
Awesome, this looks great 💯 

Got a sample shrink/outcome dump I can see? 😀

**Comment by moodmosaic** [comment]
Sure :smiley:

```
> Gen.printSample Gen.dateTime;;

=== Outcome ===
7/18/8130 12:36:13 AM
=== Shrinks ===
7/18/2000 12:36:13 AM
7/18/5065 12:36:13 AM
7/18/6598 12:36:13 AM
7/18/7364 12:36:13 AM
7/18/7747 12:36:13 AM
7/18/7939 12:36:13 AM
7/18/8035 12:36:13 AM
7/18/8083 12:36:13 AM
7/18/8107 12:36:13 AM
7/18/8119 12:36:13 AM
7/18/8125 12:36:13 AM
7/18/8128 12:36:13 AM
7/18/8129 12:36:13 AM
1/18/8130 12:36:13 AM
4/27/8130 12:36:13 AM
6/27/8130 12:36:13 AM
7/1/8130 12:36:13 AM
7/9/8130 12:36:13 AM
7/14/8130 12:36:13 AM
7/16/8130 12:36:13 AM
7/17/8130 12:36:13 AM
7/18/8130 12:00:13 AM
7/18/8130 12:18:13 AM
7/18/8130 12:27:13 AM
7/18/8130 12:32:13 AM
7/18/8130 12:34:13 AM
7/18/8130 12:35:13 AM
7/18/8130 12:36:00 AM
7/18/8130 12:36:07 AM
7/18/8130 12:36:10 AM
7/18/8130 12:36:12 AM
.
=== Outcome ===
5/10/4337 9:48:09 PM
=== Shrinks ===
5/10/2000 9:48:09 PM
5/10/3169 9:48:09 PM
5/10/3753 9:48:09 PM
5/10/4045 9:48:09 PM
5/10/4191 9:48:09 PM
5/10/4264 9:48:09 PM
5/10/4301 9:48:09 PM
5/10/4319 9:48:09 PM
5/10/4328 9:48:09 PM
5/10/4333 9:48:09 PM
5/10/4335 9:48:09 PM
5/10/4336 9:48:09 PM
1/10/4337 9:48:09 PM
3/10/4337 9:48:09 PM
4/7/4337 9:48:09 PM
5/1/4337 9:48:09 PM
5/5/4337 9:48:09 PM
5/8/4337 9:48:09 PM
5/9/4337 9:48:09 PM
5/10/4337 12:48:09 AM
5/10/4337 11:48:09 AM
5/10/4337 4:48:09 PM
5/10/4337 7:48:09 PM
5/10/4337 8:48:09 PM
5/10/4337 9:00:09 PM
5/10/4337 9:24:09 PM
5/10/4337 9:36:09 PM
5/10/4337 9:42:09 PM
5/10/4337 9:45:09 PM
5/10/4337 9:47:09 PM
5/10/4337 9:48:00 PM
5/10/4337 9:48:05 PM
5/10/4337 9:48:07 PM
5/10/4337 9:48:08 PM
.
=== Outcome ===
5/11/3490 1:48:01 AM
=== Shrinks ===
5/11/2000 1:48:01 AM
5/11/2745 1:48:01 AM
5/11/3118 1:48:01 AM
5/11/3304 1:48:01 AM
5/11/3397 1:48:01 AM
5/11/3444 1:48:01 AM
5/11/3467 1:48:01 AM
5/11/3479 1:48:01 AM
5/11/3485 1:48:01 AM
5/11/3488 1:48:01 AM
5/11/3489 1:48:01 AM
1/11/3490 1:48:01 AM
3/11/3490 1:48:01 AM
4/5/3490 1:48:01 AM
5/1/3490 1:48:01 AM
5/6/3490 1:48:01 AM
5/9/3490 1:48:01 AM
5/10/3490 1:48:01 AM
5/11/3490 12:48:01 AM
5/11/3490 1:00:01 AM
5/11/3490 1:24:01 AM
5/11/3490 1:36:01 AM
5/11/3490 1:42:01 AM
5/11/3490 1:45:01 AM
5/11/3490 1:47:01 AM
5/11/3490 1:48:00 AM
.
=== Outcome ===
4/18/1139 2:06:52 AM
=== Shrinks ===
4/18/2000 2:06:52 AM
4/18/1570 2:06:52 AM
4/18/1354 2:06:52 AM
4/18/1246 2:06:52 AM
4/18/1192 2:06:52 AM
4/18/1165 2:06:52 AM
4/18/1152 2:06:52 AM
4/18/1145 2:06:52 AM
4/18/1142 2:06:52 AM
4/18/1140 2:06:52 AM
1/23/1139 2:06:52 AM
2/8/1139 2:06:52 AM
3/23/1139 2:06:52 AM
4/1/1139 2:06:52 AM
4/9/1139 2:06:52 AM
4/14/1139 2:06:52 AM
4/16/1139 2:06:52 AM
4/17/1139 2:06:52 AM
4/18/1139 12:06:52 AM
4/18/1139 1:06:52 AM
4/18/1139 2:00:52 AM
4/18/1139 2:03:52 AM
4/18/1139 2:05:52 AM
4/18/1139 2:06:00 AM
4/18/1139 2:06:26 AM
4/18/1139 2:06:39 AM
4/18/1139 2:06:46 AM
4/18/1139 2:06:49 AM
4/18/1139 2:06:51 AM
.
=== Outcome ===
9/9/3598 1:03:57 AM
=== Shrinks ===
9/9/2000 1:03:57 AM
9/9/2799 1:03:57 AM
9/9/3199 1:03:57 AM
9/9/3399 1:03:57 AM
9/9/3499 1:03:57 AM
9/9/3549 1:03:57 AM
9/9/3574 1:03:57 AM
9/9/3586 1:03:57 AM
9/9/3592 1:03:57 AM
9/9/3595 1:03:57 AM
9/9/3597 1:03:57 AM
1/14/3598 1:03:57 AM
5/14/3598 1:03:57 AM
7/14/3598 1:03:57 AM
8/14/3598 1:03:57 AM
9/1/3598 1:03:57 AM
9/5/3598 1:03:57 AM
9/7/3598 1:03:57 AM
9/8/3598 1:03:57 AM
9/9/3598 12:03:57 AM
9/9/3598 1:00:57 AM
9/9/3598 1:02:57 AM
9/9/3598 1:03:00 AM
9/9/3598 1:03:29 AM
9/9/3598 1:03:43 AM
9/9/3598 1:03:50 AM
9/9/3598 1:03:54 AM
9/9/3598 1:03:56 AM
.
val it : unit = ()
> 
```

**Pull request merged**